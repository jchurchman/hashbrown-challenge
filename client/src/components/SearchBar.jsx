import { useState } from 'react'
import styled from 'styled-components'
import {
  TextField,
  InputAdornment,
} from '@material-ui/core'
import SearchIcon from '@material-ui/icons/Search';

const SearchBar = ({ handleFormSubmit }) => {
  const [text, setText] = useState('')
  const handleSubmit = e => {
    e.preventDefault()
    console.log('text is ', text)
    handleFormSubmit(text)
    setText('')
  }

  const handleChange = e => {
    setText(e.target.value)
  }
  return (
    <form onSubmit={handleSubmit}>
      <TextField
        onChange={handleChange}
        value={text}
        name="search"
        helperText="Enter a hashtag and press the 'return' key to see tweets that relate."
        required
        type="search"
        InputProps={{
          startAdornment: (
            <InputAdornment position="start">
              {'#'}
            </InputAdornment>
          ),
          endAdornment : (
            <InputAdornment position="end">
              <SearchIcon />
            </InputAdornment>
          )
        }}
      />
    </form>
  )
}

export default SearchBar