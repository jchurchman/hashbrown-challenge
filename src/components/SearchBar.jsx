import { useState, useCallback } from 'react'
import {
  TextField,
  InputAdornment,
} from '@material-ui/core'
import SearchIcon from '@material-ui/icons/Search';

const SearchBar = ({ handleFormSubmit }) => {
  const [text, setText] = useState('')

  const handleSubmit = useCallback(e => {
    e.preventDefault()

    handleFormSubmit(text)
    setText('')
  }, [text, handleFormSubmit])

  const handleChange = useCallback(e => {
    setText(e.target.value)
  }, [])

  return (
    <form onSubmit={handleSubmit}>
      <TextField
        onChange={handleChange}
        value={text}
        name="search"
        helperText="Enter a hashtag and press the 'return' key to see tweets that relate."
        required
        type="search"
        InputProps={{
          startAdornment: (
            <InputAdornment position="start">
              {'#'}
            </InputAdornment>
          ),
          endAdornment : (
            <InputAdornment position="end">
              <SearchIcon />
            </InputAdornment>
          )
        }}
      />
    </form>
  )
}

export default SearchBar