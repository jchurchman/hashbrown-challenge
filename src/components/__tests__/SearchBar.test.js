import renderer from 'react-test-renderer';
import SearchBar from '../SearchBar';

describe('SearchBar ', () => {
  it('renders as expected ', () => {
    const tree = renderer
      .create(<SearchBar handleFormSubmit={() => {}} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})