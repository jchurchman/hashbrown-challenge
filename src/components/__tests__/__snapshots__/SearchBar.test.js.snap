// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar  renders as expected  1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd"
      onClick={[Function]}
    >
      <div
        className="MuiInputAdornment-root MuiInputAdornment-positionStart"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          #
        </p>
      </div>
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch"
        disabled={false}
        name="search"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="search"
        value=""
      />
      <div
        className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </div>
    </div>
    <p
      className="MuiFormHelperText-root Mui-required"
    >
      Enter a hashtag and press the 'return' key to see tweets that relate.
    </p>
  </div>
</form>
`;
